### ISSUE TITLE
MyPaint crashes when my pen touches the graphic tablet (GAOMON S620)

### ISSUE THREAD

**Issue by smerkley** [mypaint/mypaint#1126]
MyPaint crashes when my pen touches the graphic tablet (GAOMON S620). Same issue happens when I run gtk3-demo-event-axes.exe and touch the pen to the tablet, the demo exits with status 139. The tablet works in other apps just fine (Gimp 2.10) and in Windows.

#### Basic system details

MyPaint version: MyPaint v2.0.1 (2020-05-29)
Operating system: Windows 7

#### Steps to reproduce

1. Start MyPaint
2. Touch pen to tablet

**Comment by jplloyd** [comment]
It would be helpful if you could test if the same issue occurs with the [latest alpha build](https://ci.appveyor.com/project/achadwick/mypaint/builds/37056649/artifacts) - download and extract the 7z file and run mypaint.exe

If possible, it would be nice to confirm whether or not this occurs only on Windows 7, or also on WIndows 10.

**Comment by smerkley** [comment]
Yep, same behavior on the latest alpha. No ability to test on Windows 10.

**Comment by ftrastour** [comment]
Hi,
same issue on Windows 10 withversion 2.0.1+git.478280f2 (win10 x64).

**Comment by odysseywestra** [comment]
Then looks like we are having a driver issue with upstream on GDK/GTK since it happens on the `gtk3-demo-event-axes.exe` end too. Not much we can do with mypaint so you will need to open a ticket upstream: https://gitlab.gnome.org/GNOME/gtk/

@jplloyd anything else we can do to verify its an upstream issue?

**Comment by jplloyd** [comment]
Well, running mypaint-debug.exe to get the debug-level output when the crash occurs would be good, to see if there are any clues in the output. But yeah, sounds like an upstream issue.

**Comment by JonnySmithin** [comment]
I have the same problem with the GAOMON M106K PRO tablet. When I touch the tablet with the pen, the program crashes. The pen works correctly in Krita.
MyPaint version: MyPaint v2.0.1
Operating system: Windows 7
Gaomon driver version: 16.0.0.20 (latest 2020-12-11)
![image](https://user-images.githubusercontent.com/77220900/104815661-76dcbf00-5848-11eb-89ce-0d5af71b3a14.png)

The debug-level output from mypaint-debug.exe:
![image](https://user-images.githubusercontent.com/77220900/104816055-303c9400-584b-11eb-9fe3-03b05b24d7f3.png)

Running python3.exe...
INFO: mypaint: Installation layout: conventional POSIX-like structure with prefix 'C:/Program Files/mypaint-w64/mingw64'
 
INFO: gui.main: No locale setting found, using system locale
INFO: lib.i18n: Windows: LANG='ru_RU'
INFO: lib.i18n: Windows: LANGUAGE='ru_RU'
INFO: lib.i18n: Value of LANGUAGE after cleanup: 'ru_RU'
WARNING: C:/Program Files/mypaint-w64/mingw64/lib/mypaint/lib/gettext_setup.py: No bindtextdomain builtins found in modu
le 'locale'.
INFO: C:/Program Files/mypaint-w64/mingw64/lib/mypaint/lib/gettext_setup.py: Trying platform-specific fallback hacks to
find bindtextdomain funcs.
INFO: C:/Program Files/mypaint-w64/mingw64/lib/mypaint/lib/gettext_setup.py: Windows: found working bindtextdomain funcs
 in 'libintl-8.dll' (ctypes)
INFO: gui.compatibility: Setting mode to 2.x (standard)
INFO: gui.compatibility: Setting default layer type to Pigment
INFO: gui.device: New device 'System Aggregated Pointer' (GDK_SOURCE_MOUSE, axes:2, class=GdkDeviceWin32, vendor=None, p
roduct=None)
INFO: gui.document: Initialized background from 'C:/Program Files/mypaint-w64/mingw64/share/mypaint/backgrounds/mrmamurk
/mamurk_e_1.png'
WARNING: gui.keyboard: Ignoring keybinding for '<Actions>/BrushModifierActions/BlendModeMenu'
INFO: gui.brushmanager: Switching default pigment setting to On
INFO: lib.document: load_ora: 'C:/Users/User/AppData/Local/mypaint/scratchpads/autosave.ora'
INFO: lib.document: 0.080s load_ora total
INFO: gui.filehandling: Loaded scratchpad from 'C:/Users/User/AppData/Local/mypaint/scratchpads/autosave.ora'
INFO: gui.compatibility: Setting default layer type to Pigment
INFO: gui.device: New device 'GaomonTablet Pressure Stylus' (GDK_SOURCE_PEN, axes:5, class=GdkDeviceWintab, vendor=None,
 product=None)
 
(mypaint:4632): Gdk-CRITICAL **: 21:26:47.382: gdk_device_get_source: assertion 'GDK_IS_DEVICE (device)' failed
--: line 1:   333 Segmentation fault      "$@"
python3.exe exited with status 139.
Press return to close this window.

**Comment by jplloyd** [comment]
Thank you @JonnySmithin 

Krita does not use Gtk (and GIMP still uses gtk2), so unfortunately that only tells us that the problem is either in Gtk or our usage of Gtk. The problem is similar to this issue (https://gitlab.gnome.org/GNOME/gtk/-/issues/1522) that was apparently resolved after a driver update.

Create an issue in the gtk tracker (if there isn't one already) with the full details, including that it crashes in gtk3-demo.exe.
You should probably also add a reference to the Huion issue, and a link back to this issue.

**Comment by odysseywestra** [comment]
Makes me wonder if both tablets use the same generic driver... I wonder if they install the Huion if that might work though those drivers might be looking for certain usb ids like how Linux does it.

**Comment by plgrm44** [comment]
Hello from 2021. Still not working.

**Comment by jplloyd** [comment]
> Hello from 2021. Still not working.

This issue was created in 2021. Do you mean that new drivers have been released this year (and that it still does not work with those)?

**Comment by plgrm44** [comment]
> > Hello from 2021. Still not working.
> 
> This issue was created in 2021. Do you mean that new drivers have been released this year (and that it still does not work with those)?
Sorry, i was wrong about the 2021)

**Comment by lb90** [comment]
Hello! A fix for this Wintab input issue [has been merged](https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3644), the fix will be available in the next GTK3 release (3.24.31)

**Comment by Lavember** [comment]
hello any update

**Comment by Sonlok** [comment]
Idk if anyone is still getting this, but my gaomon pd2200 did, i just install the driver for the huion kamvas pro 22 and the problem got solved

**Comment by fulmerw** [comment]
Same thing happens for me with my huion kamvas 13 gen 3

**Comment by RichyT** [comment]
Happening with PD1161. Fresh installed MyPaint 2.0.1+git.478280f2 on Windows 10. GaomonTablet v16.0.0.219

Not too worried as I intend to use this on Linux anyway. Just reporting a datapoint.

I see that the release of MyPaint is using GTK 3.24.20 and lb90 said that the fix would be available in 3.24.31 in 2021. The copyright is 2005-2020. Is this project abandoned?
